packages/admin/admin/src/form/FinalFormNumberInput.tsx

```diff
@@ -0,0 +1,94 @@
+import { InputBase, InputBaseProps } from "@mui/material";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+import { useIntl } from "react-intl";
+
+import { ClearInputAdornment } from "../common/ClearInputAdornment";
+
+export type FinalFormNumberInputProps = InputBaseProps &
+    FieldRenderProps<number> & {
+        clearable?: boolean;
+        decimals?: number;
+    };
+
+export function FinalFormNumberInput({
+    meta,
+    input,
+    innerRef,
+    clearable,
+    endAdornment,
+    decimals = 0,
+    ...props
+}: FinalFormNumberInputProps): React.ReactElement {
+    const intl = useIntl();
+
+    const [formattedNumberValue, setFormattedNumberValue] = React.useState("");
+
+    const getFormattedValue = React.useCallback(
+        (value: number | undefined) => {
+            const formattedValue =
+                value !== undefined ? intl.formatNumber(value, { minimumFractionDigits: decimals, maximumFractionDigits: decimals }) : "";
+            return formattedValue;
+        },
+        [decimals, intl],
+    );
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+        setFormattedNumberValue(value);
+    };
+
+    const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+        const numberParts = intl.formatNumberToParts(1111.111);
+        const decimalSymbol = numberParts.find(({ type }) => type === "decimal")?.value;
+        const thousandSeparatorSymbol = numberParts.find(({ type }) => type === "group")?.value;
+
+        const numericValue = parseFloat(
+            value
+                .split(thousandSeparatorSymbol || "")
+                .join("")
+                .split(decimalSymbol || ".")
+                .join("."),
+        );
+
+        const roundToDecimals = (numericValue: number, decimals: number) => {
+            const factor = Math.pow(10, decimals);
+            return Math.round(numericValue * factor) / factor;
+        };
+
+        const inputValue: number | undefined = isNaN(numericValue) ? undefined : roundToDecimals(numericValue, decimals);
+        input.onChange(inputValue);
```

Comment:
Why did you remove the `setFormattedNumberValue` line here?
This causes the displayed value not to be updated, e.g. when typing in invalid characters:

```suggestion
        input.onChange(inputValue);
        setFormattedNumberValue(getFormattedValue(inputValue));
```

https://github.com/vivid-planet/comet/assets/6264317/30ed2ca8-9748-40c3-8172-5f73166349bd

Reply:
It seems to me that the provided screencast works as intended? Should an invalid value be set in the form?

Reply:
> It seems to me that the provided screencast works as intended?

I don't think so:
- I type in `12` -> 
  - Both, the value and the displayed string are `12` ✅ 
- I type in an additional `d` -> 
  - The value is still `12` ✅ 
  - The displayed string includes the `d` 🤔 
- I then change the `1` to a `2` ->
  - The value is changed to `22` ✅ 
  - The displayed string is `22` and does not include the `d` 🤔

It is inconsistent when the invalid character is shown.

Reply:
Now I see

> I type in an additional `d` ->
> * The value is still `12` ✅
> * The displayed string includes the `d` 🤔

This is okay

> I then change the `1` to a `2` ->
>   
>   * The value is changed to `22` ✅
>   * The displayed string is `22` and does not include the `d` 🤔

This isn't. IMO the field shouldn't change as long as the user input is invalid.

Reply:
> IMO the field shouldn't change as long as the user input is invalid.

We currently have no way of showing the user, the value is invalid though. 
Other than replacing it by a valid (parsed) value on blur.

Reply:
@juliawegmayr changed this due to my suggestion in: https://github.com/vivid-planet/comet/pull/1938#discussion_r1638187484. I don't understand why it would work when calling the setState here, as it's also called in useEffect below... Maybe a bug?

@juliawegmayr could you please investigate?

Reply:
I found out that it doesn't work either way, so there is definitely another problem in the code. I tried several things to fix it, but I can't figure out the reason for this behaviour and I have no idea how to proceed.